### PR TITLE
[MRG+1] TST: Fixed rcparams `test_Issue_1713` test

### LINKED ISSUE #7281: rcparam encoding test is broken
The encoding test [test_Issue_1713](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_rcparams.py#L227-L237) (for #1713) tries to load a file encoded in UTF-32BE, and check that the `timezone` key is set to `UTC`.

The problem is it does not `use_default_template=False`, so the default rcparams are loaded, and the default for `timezone` is `UTC` already. It does pass `fail_on_error=True`, but that flag only validates known rcparams. Invalid rcparams print a message, but do not otherwise fail. Thus, _nothing_ is actually verified.

The flags can be modified, and the test rightly fails. However, to actually fix it, I'm not sure what it's trying to do. Does changing `LANG` actually have an effect on a running interpreter? Is this test actually useful at all? Should we switch to utf-8 only?

### PR DESCRIPTION
**Pull request by Kojoley**
Without the monkey-patching there is no good way to change locale on Windows (LANG env variable has no effect on it).

I also renamed the test to a meaningful name.

Closes #7281

### CODE REVIEW

File: lib/matplotlib/tests/test_rcparams.py [hunk 1]
```diff
@@ -224,16 +228,13 @@ def test_legend_edgecolor():
         yield _legend_rcparam_helper, rc_dict, target, get_func
 
 
-def test_Issue_1713():
+def test_unicode_parsing():
```
Comment (Kojoley) [inline lib/matplotlib/tests/test_rcparams.py]: Yes, but it what the #1713 title says.

Comment (QuLogic) [inline lib/matplotlib/tests/test_rcparams.py]: I think that's more a symptom of Python 2-ness, "Unicode vs str", but it's not strictly correct.

Comment (NelleV) [inline lib/matplotlib/tests/test_rcparams.py]: I much preferred the new title than the old.

**Comment by QuLogic** [comment]
Backported to v2.x via 6dbbb98350ead7ba6082027f7f06e34d20749f6f.

**Pull request merged**